src/basic/string-util.c

```diff
@@ -1383,6 +1383,28 @@ char* find_line_startswith(const char *haystack, const char *needle) {
         return p + strlen(needle);
 }
 
+char* find_line(const char *haystack, const char *needle) {
+        char *p;
+
+        assert(haystack);
+        assert(needle);
+
+        /* Finds the first line in 'haystack' that match the specified string. Returns a pointer to the
+         * next line after it */
+
+        p = find_line_startswith(haystack, needle);
+        if (!p)
+                return NULL;
+
+        if (*p != 0 && !strchr(NEWLINE, *p))
+                return NULL;
+
+        if (strchr(NEWLINE, *p))
+                p++;
+
+        return p;
```

Comment:
maybe instead of these two blocks do it like this:

```c
if (*p == 0)
        return p;
if (strchr(NEWLINE, *p))
        return p + 1;

return NULL;  
```

that avoids the duplicate strchr() call

Reply:
Done